nomad/search_endpoint.go

```diff
@@ -382,6 +385,15 @@ func getResourceIter(context structs.Context, aclObj *acl.ACL, namespace, prefix
 			return iter, nil
 		}
 		return memdb.NewFilterIterator(iter, nsCapFilter(aclObj)), nil
+	case structs.SecureVariables:
+		iter, err := store.GetSecureVariablesByPrefix(ws, prefix)
+		if err != nil {
+			return nil, err
+		}
+		if aclObj == nil {
+			return iter, nil
+		}
+		return memdb.NewFilterIterator(iter, nsCapFilter(aclObj)), nil
```

Comment:
I think this check will always currently fail because we don't have a `nsCapFilter` case for our type. Maybe we should drop that in with the job ACL check, just as we've done for the other Storage API RPCs? That way we can at least have a working build that we can bench test until we've created the ACL policy.